CONTRIBUTING.md

```diff
@@ -76,23 +77,19 @@ If you get some unexpected results or want to be sure that tests run is not affe
 rm -r .mypy_cache
 ```
 
+### Testing stubs with `stubtest`
 
-### Generating Stubs using Stubgen
+Run `bash ./scripts/stubtest.sh` to test that stubs and sources are in-line.
 
-The stubs are based on auto-generated code created by Mypy's stubgen tool (see: [the stubgen docs](https://mypy.readthedocs.io/en/stable/stubgen.html)).
-To make life easier we have a helper script that auto generates these stubs. To use it you can run:
+We have two special files to allow errors:
+1. `scripts/stubtest/allowlist.txt` where we store things that we really don't care about: hacks, django internal utility modules, things that are handled by our plugin, things that are not representable by type system, etc
+2. `scripts/stubtest/allowlist_generate.txt` where we store all errors there are right now. Basically, this is a TODO list: we need to work throught this list and fix things (or move entries to real `allowlist.txt`). In the end, ideally we can remove this file
 
-```bash
-python ./scripts/stubgen-django.py --django_version 3.2
-```
-
-You can also pass an optional commit hash as a second kwarg to checkout a specific commit, e.g.
+You might also want to disable `incremental` mode while working on `stubtest` changes.
+This mode leads to several known problems (stubs do not show up or have strange errors).
 
-```bash
-python ./scripts/stubgen-django.py --django_version 3.2 --commit_sha <commit_sha>
-```
+**Important**: right now we only run `stubtest` on Python3.9, any other versions might generate different outputs. Any work to create per-version allowlists is welcome.
```

Comment:
can you explain why 3.9? It's not the minimum or maximum version supported by Django 4.2, seems like an odd choice.

Reply:
3.9 is also the Python version we run our main tests on. But I agree, sounds arbitrary.

Shall we use the newest supported Python? It's faster as well.

Reply:
Latest sounds like a good policy for now. I think we can merge this using 3.11 and upgrade the other workflows in another PR.